app/api/data-sources/route.ts

```diff
@@ -31,6 +28,13 @@ export async function POST(request: NextRequest) {
     return NextResponse.json({ success: false, error: 'Environment ID is required' }, { status: 400 });
   }
 
+  if (
+    userAbility.cannot(PermissionAction.create, PermissionResource.DataSource) &&
+    userAbility.cannot(PermissionAction.create, subject(PermissionResource.Environment, { id: environmentId }))
```

Comment:
Do we need the read on environment or create? 
I was guessing read on environment + create on datasource?